### PR TITLE
DSPDC-284 Flesh out technical documentation.

### PR DESCRIPTION
**Pull request by danxmoran**
You can navigate around the rendered markdown [here](https://github.com/DataBiosphere/transporter/blob/dm-transporter-docs-DSPDC-284/README.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,40 +41,9 @@ sbt:transporter> test
...
-installation with default settings, the config should be:
-```bash
-$ cat <<-EOF > manager/src/main/resources/application.conf
-org.broadinstitute.transporter.db.username = "$(whoami)"
-org.broadinstitute.transporter.db.password = ""
-EOF
-```
+## Running Locally
+1. Run the [Manager](manager/README.md)
+2. Run an [agent](agents/README.md)
```
Comment (benjamincarlin) [inline README.md]: same for agents 

https://github.com/DataBiosphere/transporter/blob/dm-transporter-docs-DSPDC-284/agents/README.md#running-locally

Comment (benjamincarlin) [reply README.md]: Note that the links are for m-transporter-docs-DSPDC-284 branch

File: agents/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  failure messages, and pushes those messages onto the queue's "result" topic
+
+## Implementations
+Common agent code is captured in a [template](template/README.md). So far,
+we've produced the following concrete implementation of the template:
+* [Echo](echo/README.md) (no-op)
+* [AWS->GCP](aws-to-gcp/README.md)
+
+## Running Locally
+To run an agent locally, first run the [Manager](../manager/README.md)
```
Comment (benjamincarlin) [inline agents/README.md]: same with git hash

also maybe make this have a number/bullet point e.g: 

`Step 1. To run an agent locally, first run the [Manager](../manager/README.md)`

File: agents/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+locally as a background process.
+
+Next, initialize a queue resource in the Manager for the agent to connect to.
```
Comment (benjamincarlin) [inline agents/README.md]: Step 2.

etc

File: manager/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+| PUT | `/transfers/{request-id}/reconsider` | Reset the state of all failed transfers in a request to 'Pending'. |
+
+### Transfer-level
+Paths below prefixed with `/api/transporter/v1/queues/{name}/transfers/{request-id}`.
+
+| Verb | Path | Description |
+| ---- | ---- | ----------- |
+| GET | `/detail/{transfer-id}` | Get all information stored by the Manager about a specific transfer under a bulk request. |
+
+## Running Locally
```
Comment (benjamincarlin) [inline manager/README.md]: add bullet points/or numbers for each step would be great!

File: manager/README.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Transporter Manager
```
Comment (benjamincarlin) [inline manager/README.md]: for both the manager and agent markdowns I would link back to the transporter readMe

File: manager/db-migrations/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+2. Postgres has been installed using Homebrew
+3. Postgres is running on the default port (5432)
+4. The database to migrate is named "postgres" (Homebrew default)
+5. The username to log into the DB with matches the OS X username of the caller (Homebrew default)
+6. There is no password for the DB user (Homebrew default)
+
+If all of the above are true, this should work:
+```bash
+$ sbt
+sbt:transporter> transporter-manager-migrations/run
```
Comment (benjamincarlin) [inline manager/db-migrations/README.md]: look at how sbt documentation provides examples for commands (also look when sbt commands are used in other parts of this PR)

Comment (danxmoran) [reply manager/db-migrations/README.md]: Looks like I picked up this style from the sbt docs: https://www.scala-sbt.org/1.x/docs/sbt-by-example.html

**Pull request merged**